service.go

```diff
@@ -269,7 +286,7 @@ func (s *serviceManager) Process(env *network.Envelope) {
 }
 
 // closeDatabase closes the database.
-// It also removes the database file if contextDataPath is set to "".
+// It also removes the database file if the path is not default (i.e. testing config)
```

Comment:
Does this mean we should never use non-default db-directory when writing prod code?

Reply:
Yes, just as you would have never called setContextPath("") in production code.

Reply:
I feel this approach will cause some confusing (but no more than before) as `NewServer` is a public API, whoever is calling it might set the directory to something other than `""`. Do you think an explicit flag is better, something like `cleanOnClose bool` in `newServiceManager`? I remember Linus saying something about setting a flag to indicate whether the server is in a test or not.